architectures/decentralized/solana-client/src/app.rs

```diff
@@ -167,13 +169,39 @@ impl App {
             self.state_options.private_key.0.clone(),
             CommitmentConfig::confirmed(),
         )?;
-        let coordinator_instance =
+        let coordinator_instance_pubkey =
             psyche_solana_coordinator::find_coordinator_instance(&self.run_id);
-        let coordinator_instance_state = backend
-            .get_coordinator_instance(&coordinator_instance)
+        let coordinator_instance = backend
+            .get_coordinator_instance(&coordinator_instance_pubkey)
             .await?;
 
-        let coordinator_account = coordinator_instance_state.coordinator_account;
+        let coordinator_account = coordinator_instance.coordinator_account;
+        let coordinator_account_pubkey = coordinator_instance.coordinator_account;
+        let coordinator_client_version = String::from(
+            &backend
+                .get_coordinator_account(&coordinator_account_pubkey)
+                .await?
+                .state
+                .client_version,
```

Comment:
Should this client_version not be hardcoded into the coordinator program, and then returned by some instruction instead of being stored in some PDA, as it will be the same for every PDA of the coordinator program, and every PDA will need to be updated when the coordinator program gets updated? (see https://solana.stackexchange.com/questions/9361/return-value-from-program-function for how to return data from an instruction from the solana devs)

Reply:
> Should this client_version not be hardcoded into the coordinator program, and then returned by some instruction instead of being stored in the run PDA, as it will be the same for every run PDA of the coordinator program, and every PDA will need to be updated when the coordinator program gets updated?

IMO no - it should be okay to update the *client* version without forcing *all existing runs* to stop and download new code.
Maybe we want a *second* version for that, but this version number is "what version of the client code do i need to run to join this", not "is this coordinator ABI-compatible with the previous version", right?

Reply:
I was just voicing some concern that if we are building a mechanism to ensure you're running the right client version anyway, we might as well also check compatibility with the coordinator program version. But @arilotter 's point taken that the purpose of this is just to make sure all the clients in a single run are running the right version - didn't realize that was the purpose before. If that's the purpose, what's currently here is fine, but I still have the above concerns.